x-pack/plugin/core/src/main/java/org/elasticsearch/snapshots/SourceOnlySnapshotRepository.java

```diff
@@ -146,8 +155,16 @@ protected void closeInternal() {
             }, Store.OnClose.EMPTY);
             Supplier<Query> querySupplier = mapperService.hasNested() ? Queries::newNestedFilter : null;
             // SourceOnlySnapshot will take care of soft- and hard-deletes no special casing needed here
-            SourceOnlySnapshot snapshot = new SourceOnlySnapshot(overlayDir, querySupplier);
-            snapshot.syncSnapshot(snapshotIndexCommit);
+            SourceOnlySnapshot snapshot;
+            snapshot = new SourceOnlySnapshot(overlayDir, querySupplier);
+            try {
+                snapshot.syncSnapshot(snapshotIndexCommit);
+            } catch (NoSuchFileException | CorruptIndexException | FileAlreadyExistsException e) {
+                logger.warn(() -> new ParameterizedMessage(
```

Comment:
This is just best effort obviously, but I think it will reliably catch the most important case of a half-completed dangling sync in the `_snapshot` directory. Now that we have sequence number based snapshotting as well the tracking directory that handles some files via hard linking I think this is probably good enough and we don't need any more sophisticated "repair" logic for this directory to make this cheaper to execute.